src/main/java/net/dries007/tfc/common/recipes/AddBaitRecipe.java

```diff
@@ -0,0 +1,92 @@
+/*
```

Comment:
This feels very awful, yet I see why it's nessecary. What I would *like* to do here is simply add an item stack modifier for "add_bait_to_rod", throw it in a advanced shapeless recipe, and call it a day.

but there would be no way to access the bait.

so here's my very hacky yet genius idea, based on the idea of `ForgeHooks.getCraftingPlayer()`

- In `AdvancedShapedRecipe` and `AdvancedShapelessRecipe`, just before the call to `assemble`, call `RecipeHelpers.setCraftingContainer()`:
- This sets a `ThreadLocal<CraftingContainer>`
- Then in a custom `AttachBaitToRodModifier` implementation, call `RecipeHelpers.getCraftingContainer()`
- Use this to obtain the "bait" item (the rod item will be given as the input)
- Profit.

```java
private static final ThreadLocal<CraftingContainer> CRAFTING_CONTAINER = ThreadLocal.withInitial(() -> null);

/**
 * @return The crafting container of the player currently crafting an recipe which might need it. Provided by {@link AdvancedShapedRecipe} and {@link AdvancedShapelessRecipe}. Used by {@link AddBaitToRodModifier}.
 */
@Nullable
public static CraftingContainer getCraftingContainer()
{
	return CRAFTING_CONTAINER.get();
}

public static void setCraftingContainer(CraftingContainer container)
{
	CRAFTING_CONTAINER.set(container);
}
```